lib/mongo/socket.rb

```diff
@@ -194,7 +196,7 @@ def unix_socket?(sock)
     def set_socket_options(sock)
       sock.set_encoding(BSON::BINARY)
 
-      unless unix_socket?(sock) && BSON::Environment.jruby?
+      if timeout && !(unix_socket?(sock) && BSON::Environment.jruby?)
         encoded_timeout = [ timeout, 0 ].pack(TIMEOUT_PACK)
         sock.setsockopt(SOL_SOCKET, SO_RCVTIMEO, encoded_timeout)
```

Comment:
Since we're now using #read_nonblock, I removed this. I don't see that it would have any effect.